core/src/main/java/org/apache/iceberg/encryption/StandardEncryptionManager.java

```diff
@@ -67,7 +72,15 @@ public InputFile decrypt(EncryptedInputFile encrypted) {
   @Override
   public Iterable<InputFile> decrypt(Iterable<EncryptedInputFile> encrypted) {
     // Bulk decrypt is only applied to data files. Returning source input files for parquet.
-    return Iterables.transform(encrypted, this::decrypt);
+    if (nativeDataEncryption) {
```

Comment:
There is no way to know the intended file format or whether it will use native encryption at this point. I think this needs to always return the result of `decrypt`.